sqlglot/optimizer/qualify_columns.py

```diff
@@ -599,6 +599,33 @@ def _expand_struct_stars(
     return new_selections
 
 
+def _expand_struct_stars_risingwave(expression: exp.Dot) -> t.List[exp.Alias]:
+    """[RisingWave] Expand/Flatten (foo.bar).*, where bar is a struct column"""
+    parens = t.cast(exp.Paren, expression.find(exp.Paren))
+
+    # check that last selected identifier is struct for nested structs
+    if not parens.this.is_type(exp.DataType.Type.STRUCT):
```

Comment:
I suggest avoiding the cast here and ensuring `parens` is an instance of `Paren` before assuming `this` exists. If it's either not a `Paren`, or the type isn't a struct, return the empty list.